src/misc.c

```diff
@@ -162,15 +162,13 @@ char *fgetll(char *s, size_t size, FILE *stream)
 			ungetc(c, stream);
 		return s;
 	}
-	cp = s;
+	cp = strdup(s);
+
 	while (1) {
-		cp2 = cp;
-		cp = mem_alloc(len+60000);
-		strcpy(cp, cp2);
-		if (cp2 != s)
-			MEM_FREE(cp2);
+		cp = realloc(cp, 2 * len);
+
```

Comment:
Yeah, I will change that. I just needed a quick fix for testing huge (really huge) RAR files. The original code took several minutes to reach a couple hundred megabytes (out of 2 GB) and just kept getting exponentially slower so it would never finish...